### PR TITLE
Fixes #1166 &  #1168: Added robolectric test for Topic-Revision Fragment

### LINKED ISSUE #1166: Topic-revision - Tablet (Landscape) (Lowfi)
Mocks: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/d1520abf-1688-4d47-9fdd-819f5b37d3db/L-Topic-Page-Revision-Tab

Implement low-fi UI for **Topic-revision** tablet landscape mode

**Target PR date**: 23 June 2020
**Target completion date**: 27 June 2020

### LINKED ISSUE #1168: Topic-revision - Tablet (Portrait) (Lowfi)
Mocks: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/0e461d40-e024-4d56-9ae5-9697b1bbf450/P-Topic-Page-Revision-Tab

Implement low-fi UI for **Topic-revision** tablet portrait mode

**Target PR date**: 23 June 2020
**Target completion date**: 27 June 2020

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation

- Added Robolectric tests for TopicRevisionFragment
- Updated the `span count` for tablet devices in TopicRevisionFragment
- Fixes #1166 
- Fixes #1168 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@MohamedMedhat1998 I am able to understand why this PR has been created separately because as per my understanding these test cases should be introduced in low-fi/implementation PR and not in a separate PR. 

Reason behind above explanation:
This branch is on develop and if this gets approved and merged, then when we merge lowfi code we cannot be sure if these test cases still pass and as we do not enforce these test case checks on Github Actions right now, we might end up including incorrect test cases or incorrect implementation in `develop`.

**Comment by MohamedMedhat1998** [comment]
> 
> 
> @MohamedMedhat1998 I am able to understand why this PR has been created separately because as per my understanding these test cases should be introduced in low-fi/implementation PR and not in a separate PR.
> 
> Reason behind above explanation:
> This branch is on develop and if this gets approved and merged, then when we merge lowfi code we cannot be sure if these test cases still pass and as we do not enforce these test case checks on Github Actions right now, we might end up including incorrect test cases or incorrect implementation in `develop`.

It was done in this way because if I wrote test cases for low-fi landscape mode in one PR and low-fi portrait mode in another PR, there will be a conflict when merging.  
So, here I have two options: 

1. Avoiding the conflict by making the low-fi for both landscape and portrait in one PR (not recommended as you mentioned)
2. Let the conflict happen then resolve it  

@rt4914 what option should I stick with?

**Comment by MohamedMedhat1998** [comment]
@rt4914 I have merged #1455 & #1454 into this PR so that #1166 & #1168 will be closed by this PR.  
I'm closing the other PRs.  
PTAL

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -153,4 +154,4 @@
       android:name=".walkthrough.WalkthroughActivity"
       android:theme="@style/OppiaThemeWithoutActionBar" />
   </application>
-</manifest>
+</manifest>
```
Comment (rt4914) [inline app/src/main/AndroidManifest.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivity.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package org.oppia.app.testing
+
+import android.os.Bundle
+import org.oppia.app.activity.InjectableAppCompatActivity
+import org.oppia.app.topic.RouteToRevisionCardListener
+import org.oppia.app.topic.revisioncard.RevisionCardActivity
+import javax.inject.Inject
+
+class TopicRevisionTestActivity : InjectableAppCompatActivity(), RouteToRevisionCardListener {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivity.kt:11]: Add Kdoc for this activity.

File: app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivityPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import javax.inject.Inject
+
+@ActivityScope
+class TopicRevisionTestActivityPresenter @Inject constructor(
+  private val activity: AppCompatActivity
+) {
+
+  fun handleOnCreate() {
+    activity.setContentView(R.layout.activity_topic_revision_test)
+    val topicRevisionFragment = TopicRevisionFragment.newInstance(0, "")
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivityPresenter.kt]: Nit:
```
    val topicRevisionFragment =
      TopicRevisionFragment.newInstance(internalProfileId = 0, topicId = "")
```

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt [hunk 4]
```diff
@@ -14,6 +14,7 @@ import javax.inject.Inject
 /** Fragment that card for topic revision. */
 class TopicRevisionFragment : InjectableFragment(), RevisionSubtopicSelector {
   companion object {
+    const val TOPIC_REVISION_FRAGMENT_TAG = "TOPIC_REVISION_FRAGMENT_TAG"
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Nit:
`internal const val `

File: app/src/main/res/layout/activity_topic_revision_test.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rt4914) [inline app/src/main/res/layout/activity_topic_revision_test.xml]: Rename this file to `topic_revision_test_activity`

File: app/src/test/java/org/oppia/app/testing/TopicRevisionSpanTest.kt [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    val recyclerView =
+      revisionFragment?.view?.findViewWithTag<RecyclerView>(recyclerviewTag)
+    return (recyclerView?.layoutManager as GridLayoutManager).spanCount
+  }
+
+  @Test
+  fun testTopicRevisionFragmentRecyclerView_hasCorrectSpanCount() {
+    launch(TopicRevisionTestActivity::class.java).use { scenario ->
+      scenario.onActivity { activity ->
+        Truth.assertThat(getRevisionSpanCount(activity)).isEqualTo(2)
```
Comment (rt4914) [inline app/src/test/java/org/oppia/app/testing/TopicRevisionSpanTest.kt]: Nit:
`assertThat(getRevisionSpanCount(activity)).isEqualTo(2)`

Here and below, and optimise imports accordingly.

**Review by rt4914 — CHANGES_REQUESTED** [review]
LGTM, suggested some nit changes.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (MohamedMedhat1998) [reply app/src/main/AndroidManifest.xml]: Done

File: app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivity.kt (hunk 2 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivity.kt:11]: Done

File: app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivityPresenter.kt (hunk 3 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/testing/TopicRevisionTestActivityPresenter.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt (hunk 4 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Done

File: app/src/main/res/layout/activity_topic_revision_test.xml (hunk 5 above)
Comment (MohamedMedhat1998) [reply app/src/main/res/layout/activity_topic_revision_test.xml]: Done

File: app/src/test/java/org/oppia/app/testing/TopicRevisionSpanTest.kt (hunk 6 above)
Comment (MohamedMedhat1998) [reply app/src/test/java/org/oppia/app/testing/TopicRevisionSpanTest.kt]: Done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**